simpleclient_spring_boot/src/test/java/io/prometheus/client/spring/boot/AutoConfigurationTests.java

```diff
@@ -0,0 +1,74 @@
+package io.prometheus.client.spring.boot;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.security.SecurityProperties;
+import org.springframework.boot.context.embedded.LocalServerPort;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.context.ApplicationContext;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.web.client.RestTemplate;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+
+/**
+ * @author Stuart Williams (pidster)
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = RANDOM_PORT)
+public class AutoConfigurationTests {
+
+    private static final Logger logger = LoggerFactory.getLogger(AutoConfigurationTests.class);
+
+    @Autowired
+    ApplicationContext context;
+
+    @Autowired
+    SecurityProperties securityProperties;
+
+    @LocalServerPort
+    int port;
+
+    @Value("${endpoints.prometheus.path}")
+    String prometheusPath;
```

Comment:
@brian-brazil @arnostv I think I'm going to close this PR and break it up into smaller units. There's too much going on and it's too complex to debate the details of each thing in a single PR thread.